module-system/module-implementations/sov-blob-storage/src/capabilities.rs

```diff
@@ -0,0 +1,57 @@
+use sov_modules_api::capabilities::BlobSelector;
+use sov_modules_api::{Context, Spec};
+use sov_rollup_interface::da::BlobReaderTrait;
+use sov_state::WorkingSet;
+
+use crate::BlobStorage;
+
+impl<C: Context> BlobSelector for BlobStorage<C> {
+    type Context = C;
+
+    fn get_blobs_for_this_slot<'a, I, B>(
+        &self,
+        current_slot: u64,
+        current_blobs: I,
+        pull_from_slot: u64,
```

Comment:
Question: I had assumed that the blob_selector would be responsible for this internally - it seems like whatever is invoking it won't have enough context to make very smart decisions here. How did you envision this being used?

Reply:
That's good point.

I think this is related to the question about dependencies. Now I realize, that it is better to encapsulated logic in blob storage.

This means following:

1. `BlobStorage` depends on `SequencerRegistry`, as in this PR. For getting preferred sequencer
2. `BlobStorage` depends on `ChainState` module, so it can pull slot number.
3. `BlobStorage` has "number of blocks to defer" as state variable, which is defined during genesis. How it is changed can be defined later.
4. `BlobStorage` implements capability `BlobSelector`, which only accepts `current_slots` and `working_set`. So completely different logic can be done using different blob selector.